spec/models/vacancy_spec.rb

```diff
@@ -312,17 +312,34 @@
 
   context 'scopes' do
     describe '#applicable' do
-      it 'finds current vacancies' do
-        expired = build(:vacancy, :expired)
-        expired.send :set_slug
-        expired.save(validate: false)
-        expires_today = create(:vacancy, expires_on: Time.zone.today)
-        expires_future = create(:vacancy, expires_on: 3.months.from_now)
+      context 'when expiry time not given' do
+        it 'finds current vacancies' do
+          expired = build(:vacancy, :expired)
+          expired.send :set_slug
+          expired.save(validate: false)
+          expires_today = create(:vacancy, expires_on: Time.zone.today)
+          expires_future = create(:vacancy, expires_on: 3.months.from_now)
+
+          results = Vacancy.applicable
+          expect(results).to include(expires_today)
+          expect(results).to include(expires_future)
+          expect(results).to_not include(expired)
+        end
+      end
 
-        results = Vacancy.applicable
-        expect(results).to include(expires_today)
-        expect(results).to include(expires_future)
-        expect(results).to_not include(expired)
+      context 'when expiry time given' do
+        it 'finds current vacancies' do
+          expired_earlier_today = build(:vacancy, expires_on: Time.zone.today,
```

Comment:
This could be made easier to read i.e. rather than defining one expired vacancy, doing some actions on it and then defining the second one. Maybe define both before doing the actions?

e.g `expired_earlier_today` and `expires_later_today` could be built/created in the context block using `let`.

Reply:
Resolved in 3d78ebf